src/DotVVM.Framework.StartupPerfTests/StartupPerformanceTest.cs

```diff
@@ -0,0 +1,153 @@
+
+                for (var i = 0; i < repeat; i++)
+                {
+                    TraceOutput($"Attempt #{i + 1}");
+                    measuredTime += RunProcessAndWaitForHealthCheck(@"C:\Program Files (x86)\IIS Express\iisexpress.exe", $@"""/path:{dir}"" /port:{port}", dir, urlToTest);
+                }
+            }
+            else if (type == TestTarget.AspNetCore)
+            {
+                // ASP.NET Core
+                TraceOutput($"Publishing...");
+                RunProcessAndWait(@"dotnet", $@"publish -c Release -o ""{tempDir}""", dir);
+
+                for (var i = 0; i < repeat; i++)
+                {
+                    TraceOutput($"Attempt #{i + 1}");
+                    measuredTime += RunProcessAndWaitForHealthCheck(@"dotnet", $@"""./{Path.GetFileNameWithoutExtension(projectPath)}.dll"" --urls {urlToTest}", tempDir, urlToTest);
+                }
+            }
+            else
+            {
+                throw new NotSupportedException();
+            }
+
+            TraceOutput($"Average time: {measuredTime / repeat}");
+
+            TraceOutput($"Removing temp dir...");
+            FileSystemHelper.RemoveDir(tempDir);
+            TraceOutput($"Done");
+        }
+
+        private void RunProcessAndWait(string path, string arguments, string workingDirectory)
+        {
+            TraceOutput($"Running {path} {arguments}");
+
+            var psi = new ProcessStartInfo(path, arguments) {
+                CreateNoWindow = true,
+                UseShellExecute = false,
+                WindowStyle = ProcessWindowStyle.Hidden,
+                WorkingDirectory = workingDirectory
+            };
+            var process = Process.Start(psi);
+            process.WaitForExit();
+            if (process.ExitCode != 0)
+            {
+                throw new Exception($"Process exited with code {process.ExitCode}!");
+            }
+        }
+
+        private long RunProcessAndWaitForHealthCheck(string path, string arguments, string workingDirectory, string urlToTest)
+        {
+            var psi = new ProcessStartInfo(path, arguments) {
+                CreateNoWindow = true,
+                UseShellExecute = false,
+                WindowStyle = ProcessWindowStyle.Hidden,
+                WorkingDirectory = workingDirectory
+            };
+            var process = Process.Start(psi);
+
+            var sw = new Stopwatch();
+            sw.Start();
+
+            retry:
+            try
+            {
+                var wc = new WebClient();
+                var response = wc.DownloadString(urlToTest);
+            }
+            catch (WebException ex) when (ex.InnerException is HttpRequestException hrex && (hrex.InnerException is IOException || hrex.InnerException is SocketException))
+            {
+                Thread.Sleep(100);
+                goto retry;
```

Comment:
![image](https://user-images.githubusercontent.com/7894687/84760954-48e48300-afb8-11ea-91d2-ad494c5a8d81.png)

:slightly_smiling_face:

Reply:
Added the timeout option (with a default value of 10 seconds). If the HTTP port doesn't open within the timeout, an exception is thrown.